plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/CachingJdbcClient.java

```diff
@@ -412,28 +412,28 @@ public void setTableComment(ConnectorSession session, JdbcTableHandle handle, Op
     public void setColumnComment(ConnectorSession session, JdbcTableHandle handle, JdbcColumnHandle column, Optional<String> comment)
     {
         delegate.setColumnComment(session, handle, column, comment);
-        invalidateColumnsCache(handle.asPlainTable().getSchemaTableName());
+        invalidateTableCaches(handle.asPlainTable().getSchemaTableName());
```

Comment:
cmt msg

> Invalidate caches more coarsely
> 
> Table handle contains column handles. When column handle is changed,
data in cache for table handles is outdated.

I'd suggest title like "Invalidate table handle cache when column changed"

then drop "handle" in "When column handle is changed," -- it's not _the column handle_ what's changing (column handles are immutable), it's the column itself

Reply:
BTW why having stale `io.trino.plugin.jdbc.JdbcColumnHandle#comment` value inside a `JdbcTableHandle.columns` matters?

I don't think it's used explicitly. If it's used implicitly (via equals), then maybe we have concurrency problem.
i.e. what happens if one query pulls a `JdbcTableHandle` and then (before the first query planning finishes), some other query performs `setColumnComment`.
The `CachingJdbcClient` state will be eventually consistent, but the first query will be planning on an inconsistent state.

Reply:
I think the only reasonable usage of the value of the comment is in `JdbcColumnHandle.getRetainedSizeInBytes()`

How the situation described above differs from the e.g. `addColumn`?

Reply:
> I think the only reasonable usage of the value of the comment is in `JdbcColumnHandle.getRetainedSizeInBytes()`

so the invalidation doesn't matter.



> How the situation described above differs from the e.g. `addColumn`?

Good question.
Generally the JDBC table handle won't carry columns until they are projected.
https://github.com/trinodb/trino/blob/5579e701e126051ec34d50ba37d9d586a1015263/plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/BaseJdbcClient.java#L256

But I can imagine some JDBC connector eagerly populating the `JdbcTableHandle.columns` field within `io.trino.plugin.jdbc.JdbcClient#getTableHandle(io.trino.spi.connector.ConnectorSession, io.trino.spi.connector.SchemaTableName)` call. Then, `addColumn` changes the state in a way that would matter.

Reply:
anyway, I am fine with doing `invalidateTableCaches` here, for consistency.
i just hope this is "for consistency" and not "a fix"

Reply:
> JdbcTableHandle.columns matters?

It matters. Plenty of connectors for `getColumns` do a short-circuit and return columns from table handle. If they do not matter we should remove them.